src/main/java/org/sagebionetworks/web/client/widget/entity/EntityBadgeViewImpl.java

```diff
@@ -192,7 +192,13 @@ public void setEntity(final EntityHeader entityHeader) {
     entityAnchor.setPaddingBottom(0);
     entityTypeIconContainer.setWidget(entityTypeIcon);
     entityTypeIcon.setType(getEntityTypeForClassName(entityHeader.getType()));
-    entityContainer.add(entityAnchor);
+    if (entityHeader.getName().length() > 40) {
```

Comment:
Does this work as expected at a narrow screen size? Wonder if there are downsides to just giving all entities a tooltip.

Reply:
I think so based on testing, but this is not a good solution for browsing on mobile